### PR TITLE
section name & buttons fixed

### PR DESCRIPTION
**Pull request by sjurate**

### CODE REVIEW

**Review by NeHerkus — APPROVED** [review]

**Review by bridzius — APPROVED** [review]

File: src/app/application-form/preview/preview.component.ts [hunk 1]
```diff
@@ -36,10 +44,23 @@ export class PreviewComponent {
...
+    this.applicationFormService
+      .postFormData(this.data)
+      .pipe(
+        tap((res: HttpResponse<any>) => {
+          if (res.status === 200) {
+            this.router.navigate(['submitted']);
+          }
+        }),
+        catchError((error) => {
+          console.log('Error: ' + error);
```
Comment (bridzius) [inline src/app/application-form/preview/preview.component.ts]: Would be great if this was removed before getting into main.

File: src/app/application-form/preview/preview.component.ts [hunk 2]
```diff
@@ -36,10 +44,23 @@ export class PreviewComponent {
...
-    this.dialogRef.close(true);
-    this.applicationFormService.postFormData(this.data).subscribe((res) => {
-      console.log(res);
-      this.router.navigate(['submitted']);
-    });
+    this.applicationFormService
+      .postFormData(this.data)
+      .pipe(
+        tap((res: HttpResponse<any>) => {
+          if (res.status === 200) {
```
Comment (bridzius) [inline src/app/application-form/preview/preview.component.ts]: Doesn't the Observable ignore everything and just jump to `catchError` on failure? Is this check needed?

File: src/app/error/error.component.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { Component, Inject } from '@angular/core';
+import { MAT_DIALOG_DATA } from '@angular/material/dialog';
+
+@Component({
+  selector: 'app-error',
+  templateUrl: './error.component.html',
+  styleUrls: ['./error.component.scss'],
+})
+export class ErrorComponent {
+  constructor(@Inject(MAT_DIALOG_DATA) public data: any) {}
```
Comment (bridzius) [inline src/app/error/error.component.ts:10]: Refrain from using `any` if possible and instead use actual data types.

File: src/app/services/application-form-service.service.ts [hunk 4]
```diff
@@ -19,6 +19,8 @@ export class ApplicationFormService {
 
   postFormData(applicationFormParameters: ApplicationFormValues) {
     console.log('posting...');
-    return this.httpClient.post(this.apiURL, applicationFormParameters);
+    return this.httpClient.post(this.apiURL, applicationFormParameters, {
+      observe: 'response',
```
Comment (bridzius) [inline src/app/services/application-form-service.service.ts:23]: This shouldn't be needed if you added a response type generic to the `.post<>`

**Review by bridzius — APPROVED** [review]

**Pull request merged**